src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -1352,4 +1353,12 @@ std::add_rvalue_reference_t<T> declval() noexcept;
 // handled.
 bool IEEE_subnormal_handling_OK();
 
+//----------------------------------------------------------------------------------------------------
+// C2 limits
+
+// To avoid unbounded RegMask growth, we need to set a limit on the number of
+// stack slots used by BoxLockNodes. We reach this limit by deeply nesting
+// synchronized statements in Java.
+const int BoxLockNode_slot_limit = 200;
+
```

Comment:
This definition seems too C2-specific to be put in this shared file, could it be moved e.g. to `optoreg.hpp`?

Reply:
Thanks, I was unsure where to put this definition. It doesn't really relate to `OptoReg` and is rather a limitation for `RegMask`s, so I now simply put it as a constant in `regmask.hpp`.